src/ErrorHandler.ts

```diff
@@ -0,0 +1,59 @@
+import * as httpStatus from 'http-status';
+import { ErrorRequestHandler } from 'express';
+import { Logger } from './Logger';
+
+interface ErrorResponseBody {
+    error: string;
+    error_description?: string;
+    error_uri?: string;
+}
+export type RequestError = any;
+
+
+export function ErrorHandler(logger: Logger): ErrorRequestHandler {
+
+    return (err: RequestError, req, res, next) => {
+        if (err) {
+            const responseBody: ErrorResponseBody = {
+                error: 'server_error',
+            };
+            const errorContext: any = {
+                "error_message": err.message,
+                "error_stack": err.stack,
+            };
+
+            if (err.status) {
+                res.statusCode = Number(err.status);
+            }
+            if (!res.statusCode || res.statusCode < 400) {
+                res.statusCode = httpStatus.INTERNAL_SERVER_ERROR;
+            }
+
+            if (res.statusCode === httpStatus.UNAUTHORIZED) {
+                if (err.realm && err.realm.type && err.realm.env && err.realm.charset) {
```

Comment:
Juste pour info, ça peut être condensé:

```suggestion
                if (err?.realm?.type && err.realm.env && err.realm.charset) {
```